iothub/device/src/TransientFaultHandling/AsyncExecution.cs

```diff
@@ -1,125 +1,104 @@
-// Copyright (c) Microsoft. All rights reserved.
+﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
-//Copyright(c) Microsoft.All rights reserved.
-//Microsoft would like to thank its contributors, a list
-//of whom are at http://aka.ms/entlib-contributors
 
 using System;
-using System.Globalization;
+using System.Diagnostics;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.Azure.Devices.Client.TransientFaultHandling.Properties;
 
-//Licensed under the Apache License, Version 2.0 (the "License"); you
-//may not use this file except in compliance with the License. You may
-//obtain a copy of the License at
-
-//http://www.apache.org/licenses/LICENSE-2.0
-
-//Unless required by applicable law or agreed to in writing, software
-//distributed under the License is distributed on an "AS IS" BASIS,
-//WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
-//implied. See the License for the specific language governing permissions
-//and limitations under the License.
-
-// THIS FILE HAS BEEN MODIFIED FROM ITS ORIGINAL FORM.
-// Change Log:
-// 9/1/2017 jasminel Renamed namespace to Microsoft.Azure.Devices.Client.TransientFaultHandling.
-// 3/11/2020 drwill Misc code cleanup for code style and fxcopy warnings
-
 namespace Microsoft.Azure.Devices.Client.TransientFaultHandling
 {
     /// <summary>
-    /// Provides a wrapper for a non-generic <see cref="System.Threading.Tasks.Task" /> and calls into the pipeline
-    /// to retry only the generic version of the <see cref="System.Threading.Tasks.Task" />.
+    /// Runs an async task with retry policy.
     /// </summary>
-    internal class AsyncExecution : AsyncExecution<bool>
+    internal static class AsyncExecution
     {
-        private static Task<bool> cachedBoolTask;
+        private static readonly TimeSpan s_minimumTimeBetweenRetries = TimeSpan.FromSeconds(1);
 
-        public AsyncExecution(
```

Comment:
No need to use a ctor and store local members, because all the variables can be stored in the single method.